core/src/main/java/com/linecorp/armeria/server/annotation/AnnotatedService.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.annotation;
+
+import java.lang.reflect.Method;
+
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.server.HttpService;
+import com.linecorp.armeria.server.Route;
+import com.linecorp.armeria.server.ServerBuilder;
+
+/**
+ * An {@link HttpService} which is defined by a {@link Path} or HTTP method annotations.
+ * This class is designed to provide a common interface for {@link AnnotatedService}
+ * internally. Please check out the documentation at
+ * <a href="https://armeria.dev/docs/server-annotated-service">Annotated HTTP Service</a> to use this.
+ */
+public interface AnnotatedService extends HttpService {
+
+    /**
+     * Returns service name for this {@link AnnotatedService}.
+     */
+    String serviceName();
+
+    /**
+     * Returns method name which is annotated in {@link AnnotatedService}.
+     */
+    String methodName();
+
+    /**
+     * Returns the annotated service object specified with {@link ServerBuilder#annotatedService(Object)}.
+     */
+    Object serviceObject();
+
+    /**
+     * Returns the annotated service object specified with {@link ServerBuilder#annotatedService(Object)}.
+     */
+    Class<?> serviceClass();
+
+    /**
+     * Returns the target {@link Method} invoked when the request is received.
+     */
+    Method method();
+
+    /**
+     * Returns a unique ID to distinguish overloaded methods.
+     * If the method is not overloaded, it should return 0.
+     */
+    int overloadId();
```

Comment:
Yes! 
I removed it from `AnnotatedService` interface, and keep it `DefaultAnnotatedService` as public.

Reply:
However, other changes are required because of this. 
In `AnnotatedDocServicePlugin`, they need to use `overloadId()`. 
Thus, i modify method signature as well. 

<img width="768" alt="image" src="https://github.com/line/armeria/assets/90125071/8b79851d-2f6c-4b11-9c3f-4c73b87bd1b4">

Reply:
Sounds good. Thanks!